gtsam_unstable/geometry/ABC.h

```diff
@@ -261,7 +300,7 @@ State<N> operator*(const G<N>& X, const State<N>& xi) {
  * the input equivariance
  */
 template <size_t N>
-Input velocityAction(const G<N>& X, const Input& u) {
+Input velocityAction(const Group<N>& X, const Input& u) {
```

Comment:
The Input type is mixing two things I believe: a mathematical object, $(\omega,0)$ where the zeros are that “virtual” input. And then a data structure that is used to store noise parameters and what not. I think the code will be a lot more clear if you separate those concerns, rename `Input` to `InputData`, and then redefine Input as simply
```
Using Input = Vector6
```
Or even just use Vector6.